test/testcafe/framework/page-objects/components/CalloutObject.js

```diff
@@ -13,22 +15,24 @@ export default class CalloutObject {
   }
 
   isWarning() {
+    if (userVariables.gen3) {
+      return this.el.hasClass('MuiAlert-calloutWarning');
+    }
     return this.el.hasClass('infobox-warning') &&
       this.el.child('[data-se="icon"]').hasClass('warning-16');
   }
 
   isError() {
-    const hasInfoBoxErrorClass = this.el.hasClass('infobox-error');
     if (userVariables.gen3) {
-      return hasInfoBoxErrorClass;
+      return this.el.hasClass('MuiAlert-calloutError');
     }
-    return hasInfoBoxErrorClass &&
+    return this.el.hasClass('infobox-error') &&
       this.el.find('[data-se="icon"]').hasClass('error-16');
   }
 
   getTextContent() {
     if (userVariables.gen3) {
-      return this.el.textContent;
+      return this.el.find(CALLOUT_CONTENT_V3).textContent;
```

Comment:
Is the `find` necessary here? The logic in the constructor [here](https://github.com/okta/okta-signin-widget/pull/3484/files#diff-4bcc86ed383d602f2dbea0af4ee097d300d074ec0c3b90f1319dedd8266197a9R10-R14) assigns the `this.el` to the current alert/callout, so I would think it should already have reference to the correct element without the need to find?

Reply:
Yes. The idea here is to filter out non-visible `span` inside callout (see span with text "warning" in the screenshot). So instead of taking `textContent ` for whole alert, we need to get it for content div only. 

<img width="700" alt="Screenshot 2023-12-12 at 17 37 23" src="https://github.com/okta/okta-signin-widget/assets/72614880/d940dea7-ab1c-44dc-8021-3a11e15432e4">

```javascript
document.querySelector('[role="alert"]').textContent
// outputs: 'warningReminder contentResend'
```

Reply:
Just spoke to Lester about this. First off, if you trace the references it looks like this specific `getTextContent()` is barely used and doesn't specifically need `textContent`. We could use `innerText`, which we use much more frequently in our tests. 

`innerText` will get us a string of all text in the alert delimited by a `\n` character and we can update all text assertions that use `.getTextContent().eql(...)` to instead use something like `.contains(....)`. 

This will get us past the issue of having to use MUI class selectors as a way of targeting the exact parent element of the text content we want

Reply:
Ok.
It was my initial idea to fix tests by replacing `.eql()` with `.contains()`, but I thought thus tests would be less accurate.

Reply:
Done. Replaced `.eql` with `.contains`
https://github.com/okta/okta-signin-widget/pull/3484/commits/9e4cb2b505c43b91257e17263bb9392dbecb3a5e